### PR TITLE
fix crlf on neo.UnitTests and set all files to not use BOM (byte order mark)

### PR DESCRIPTION
**Pull request by igormcoelho**
Change related to base formatting issue: #982

Now reformatting all unit tests... sorry @erikzhang :joy:

### CODE REVIEW

**Comment by igormcoelho** [comment]
We can discuss here a little bit before merging... this is what is being done:
- fixing all crlf
- removing random bom prefix to no_bom

We need to double check if chinese comments are still working on files. If something strange happens, we can enforce an UTF-8 bom prefix on all files.

**Comment by vncoelho** [comment]
Maybe UTF-8 would be better. What do you think, @erikzhang?

**Comment by shargon** [comment]
UTF8 is better

**Comment by igormcoelho** [comment]
Ok, situation is like this:
```
       0     190       0  UTF-8     text    ./UT_BigDecimal.cs
       0     110       0  no_bom    text    ./Ledger/UT_StorageItem.cs
       0     147       0  UTF-8     text    ./Ledger/UT_PoolItem.cs
       0     130       0  no_bom    text    ./Ledger/UT_StorageKey.cs
       0     257       0  UTF-8     text    ./Ledger/UT_MemoryPool.cs
```
I'll unify all as UTF-8, not only neo unit tests

**Comment by igormcoelho** [comment]
Everything is UTF-8 BOM now.

**Comment by erikzhang** [comment]
I think we don't need BOM.

**Comment by erikzhang** [comment]
Before we approve/merge this kind of things, don't we need a standard first?

**Comment by igormcoelho** [comment]
> I think we don't need BOM.

I think without it, we risk corrupting non ascii chars...

I'll try to copy some chinese comments from devpack,guess it will not work without bom. Probably, your editor will create with utf8, so we need to be compatible.

**Comment by shargon** [comment]
@could you check UT and non Source files?

**Comment by igormcoelho** [comment]
> Could you check UT and non Source files?

Do you think some file is missing @shargon? I'm only taking care of not corrupting .git repository... since my script is heavily recursive. But I'll check other folders here.

**Comment by shargon** [comment]
it is because you mentioned that is for ut files, but maybe we have source files with boom too

**Comment by igormcoelho** [comment]
> it is because you mentioned that is for ut files, but maybe we have source files with boom too

Ok, now the whole repo is cleaned up for bom. The thing is, on previous PR, I have already cleaned up bom for `neo/`... but after that I was thinking if we should have or not UTF-8 bom, but Erik (and Unicode standard) convinced me that it's better to not have. So, now, it's just 90 files to change.

**Comment by igormcoelho** [comment]
A quick peek on the final format:
```
      0     486       0  no_bom    text    ./neo/Ledger/MemoryPool.cs
       0      50       0  no_bom    text    ./neo/Ledger/TransactionState.cs
       0      35       0  no_bom    text    ./.travis.yml
       0      13       0  no_bom    text    ./SECURITY.md
       0      28       0  no_bom    text    ./neo.sln
       0       7       0  no_bom    text    ./NuGet.Config
       0     131       0  no_bom    text    ./README.md
       0      48       0  no_bom    text    ./neo.UnitTests/Wallets/UT_WalletAccount.cs
       0     114       0  no_bom    text    ./neo.UnitTests/Wallets/UT_KeyPair.cs
       0      32       0  no_bom    text    ./neo.UnitTests/Wallets/UT_Wallets_Helper.cs
       0     435       0  no_bom    text    ./neo.UnitTests/Wallets/UT_Wallet.cs
```

**Comment by shargon** [comment]
We should move this to the standard, as @erikzhang said, but it is a quite urgent because cause troubles between all developers, i expect that travis will check this well before any contribution on (https://github.com/neo-project/neo/pull/1056)

**Comment by erikzhang** [comment]
How to ensure that we will not commit new files with BOM in the future?

**Comment by vncoelho** [comment]
The travis will make this check Erik, after merging this the tests for bom are going to be accomplished.
Or, as you said, both could be merged together.

We need to do that for the other repos as well.

**Comment by erikzhang** [comment]
Let's close #1056, and move the travis changes to here.

**Comment by erikzhang** [comment]
Another thing is we need an empty line before the end of files.

**Comment by shargon** [comment]
@igormcoelho take a look to https://github.com/igormcoelho/neo/pull/6

**Comment by vncoelho** [comment]
image

All passed! :D

**Comment by vncoelho** [comment]
@erikzhang, can we handle this end of file newline in another PR?

**Comment by shargon** [comment]
@vncoelho We should finish all here, i prefer to do all the checks now

**Comment by shargon** [comment]
I think that dotnet format could check both, end of file, and utf without boom

https://github.com/dotnet/format/blob/master/.editorconfig

**Comment by shargon** [comment]
@igormcoelho i think that we can remove dos2unix and only use `dotnet format` https://docs.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options?view=vs-2017

**Comment by shargon** [comment]
Is confirmed that `.editorconfig` works, so we should configure only this file and travis will check these rules

**Comment by shargon** [comment]
Now, we only need to define the editorconfig, and call dotnet format in the root of the project, and this is finished. Also, visual studio should take this file.

**Comment by igormcoelho** [comment]
Very nice, I think.I worked with it before, its very powerful. Lets use it.

**Comment by shargon** [comment]
It's time to define the tabs (i prefer tab instead of spaces) and format all the files again.

**Comment by igormcoelho** [comment]
Reminds me of this PR: https://github.com/neo-project/neo/pull/857

Perhaps we can reactivate it in the future, for specific linter features.

**Comment by igormcoelho** [comment]
> It's time to define the tabs (i prefer tab instead of spaces) and format all the files again.

Usually, it's better to adopt spaces instead of tabs, but anyway is fine for me.

**Comment by igormcoelho** [comment]
Houston we have a problem... my local `dotnet format` is not consistent to the one on travis anymore... :joy: I told you... things are getting complicated. Perhaps it's dotnet version, I'll try to upgrade my local.

**Comment by igormcoelho** [comment]
@shargon I had to disable `crlf` option on dotnet, as it's not working properly... it's not writing correct \r\n unfortunately. So, it's still better to use `dos2unix` if necessary, this works properly.

**Comment by shargon** [comment]
https://github.com/dotnet/format/issues/347#issuecomment-524913901

We need the version 4.0

**Comment by shargon** [comment]
> I believe it is better to come back with the original verification with dos2unix. Otherwise we are going to need to change a bunch of lines.

If we need to change the lines is because is needed, and these lines are wrong

**Comment by shargon** [comment]
@igormcoelho Please, try again with the version 4.0 
`dotnet tool install -g dotnet-format --version 4.0.40103 --add-source https://dotnet.myget.org/F/format/api/v3/index.json `

**Comment by shargon** [comment]
> looks like these options only work for version 3.1.37+. How will we inform this to users?

The user only need to know the rules, this is a check for ensure that they fit these rules

**Comment by igormcoelho** [comment]
Besides the version `3.1.37+`, now I discovered @shargon! It was opposite, that's why it was strange... the formatter was enforcing crlf, while we are trying to remove crlf and replace by lf. It's fixed now :+1:

**Comment by igormcoelho** [comment]
oh man, just crashed my commit hahaha same file... same line,,,

**Comment by shargon** [comment]
But do you change the file for `lf`?

**Comment by igormcoelho** [comment]
Yes, I was removing dos2unix commands. Will try again :P

**Comment by igormcoelho** [comment]
Just need to remove dos2unix now and it's finished.

**Comment by shargon** [comment]
image

I think that is working well now

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
+###############################
+# Core EditorConfig Options   #
+###############################
+
+# dotnet-format requires version 3.1.37601
```
Comment (shargon) [inline .editorconfig]: Good point!

**Comment by shargon** [comment]
https://github.com/neo-project/neo/pull/1057#issuecomment-524900688

I think that we should decide the tab style, more changes yes, but is only for today

**Comment by igormcoelho** [comment]
I tried again the `tab`, it's unfeasible now... it will change everything from scratch, we need to discuss how to do this properly, at this scale.
Regarding `lf` vs `crlf`, I think `lf` is better because it's the standard ("\n") for many algorithms, including diff... that's why things break when mixing `crlf` ("\r\n"). 
Historically, windows editors have no problem with `lf`, and they even create files like that, but the opposite is more complicated to handle... the code is 99% `lf` already, so changing to `crlf` is also a huge change (and will often break on linux servers and editors).

**Comment by shargon** [comment]
Is ready to merge for me when travis finished

**Comment by erikzhang** [comment]
I don't think we can merge this kind of thing so hurriedly.

**Comment by erikzhang** [comment]
We should remove the `TODO`.

**Pull request merged**